test/kube_testing/pods/vpn-gateway-nsc.go

```diff
@@ -0,0 +1,52 @@
+package pods
+
+import (
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func VPNGatewayNSCPod(name string, node *v1.Node, env map[string]string) *v1.Pod {
+	ht := new(v1.HostPathType)
+	*ht = v1.HostPathDirectoryOrCreate
+
+	var envVars []v1.EnvVar
+	for k, v := range env {
+		envVars = append(envVars,
+			v1.EnvVar{
+				Name:  k,
+				Value: v,
+			})
+	}
+
+	pod := &v1.Pod{
+		ObjectMeta: v12.ObjectMeta{
+			Name: name,
+		},
+		TypeMeta: v12.TypeMeta{
+			Kind: "Deployment",
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
```

Comment:
You need to add NSC as init container, since it will do infinit restart in this case.

Following is example:
```
Spec: v1.PodSpec{
			RestartPolicy: v1.RestartPolicyNever,
			Containers: []v1.Container{
				{
					Name:            "alpine-img",
					Image:           "alpine:latest",
					ImagePullPolicy: v1.PullIfNotPresent,
					Command: []string{
						"tail", "-f", "/dev/null",
					},
				},
			},
			InitContainers: []v1.Container{
				{
					Name:            "nsc",
					Image:           "networkservicemesh/nsc:latest",
					ImagePullPolicy: v1.PullIfNotPresent,
					Resources: v1.ResourceRequirements{
						Limits: v1.ResourceList{
							"nsm.ligato.io/socket": resource.NewQuantity(1, resource.DecimalSI).DeepCopy(),
						},
						Requests: nil,
					},
					Env: envVars,
				},
			},
		},
```

Reply:
Right ;) Thanks will fix it.